src/main/java/org/jabref/logic/importer/SearchBasedParserFetcher.java

```diff
@@ -39,16 +41,22 @@
             return Collections.emptyList();
         }
 
-        try (InputStream stream = getUrlDownload(getURLForQuery(query)).asInputStream()) {
+        URL urlForQuery;
+        try {
+            urlForQuery = getURLForQuery(query);
+        } catch (URISyntaxException | MalformedURLException e) {
+            LoggerFactory.getLogger(this.getClass()).info("Search URL {} is malformed", query);
+            throw new FetcherException("Search URI is malformed", e);
+        }
+        try (InputStream stream = getUrlDownload(urlForQuery).asInputStream()) {
             List<BibEntry> fetchedEntries = getParser().parseEntries(stream);
 
             // Post-cleanup
             fetchedEntries.forEach(this::doPostCleanup);
 
             return fetchedEntries;
-        } catch (URISyntaxException e) {
-            throw new FetcherException("Search URI is malformed", e);
         } catch (IOException e) {
+            LoggerFactory.getLogger(this.getClass()).info("IOException at URL {}", urlForQuery.toString());
```

Comment:
Extract to constant to be consistent with other code:
`    private static final Logger LOGGER = LoggerFactory.getLogger(SearchBasedParserFetcher.java .class);`

Reply:
I'll add a comment that variables are not possible in interfaces (are they?)

Reply:
Sorry, did not see it's an interface. Variables are possibe in interface, but only public. You could extract it to a default method but I think in this case it's irrelevant